ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -204,23 +212,86 @@ func convertToContainerDef(inputCfg *adapter.ContainerConfig, ecsContainerDef *C
 }
 
-func getResourceValue(serviceName string, inputVal, ecsVal int64, option string) int64 {
-	if ecsVal == 0 {
-		return inputVal
+// returns true if the container has a healthcheck from compose or ecs params
+func hasHealthCheck(inputCfg *adapter.ContainerConfig, ecsContainerDef *ContainerDef) bool {
+	if inputCfg.HealthCheck != nil {
+		return true
+	}
+
+	if ecsContainerDef != nil && ecsContainerDef.HealthCheck != nil {
+		return true
+	}
+
+	return false
+
+}
+
+// sets the healthcheck with the ECS Params as an override
+func getHealthcheck(inputCfg *adapter.ContainerConfig, ecsContainerDef *ContainerDef) (*ecs.HealthCheck, error) {
+	healthcheck := inputCfg.HealthCheck
+	if healthcheck == nil {
+		healthcheck = &ecs.HealthCheck{}
+	}
+
+	if ecsContainerDef != nil && ecsContainerDef.HealthCheck != nil {
+		ecsParamsHealth := ecsContainerDef.HealthCheck
+
+		healthcheck.Command = resolveStringSliceResourceOverride(inputCfg.Name, healthcheck.Command, ecsParamsHealth.Command, "healthcheck command")
+		healthcheck.Interval = resolveIntPointerResourceOverride(inputCfg.Name, healthcheck.Interval, ecsParamsHealth.Interval, "healthcheck interval")
+		healthcheck.Retries = resolveIntPointerResourceOverride(inputCfg.Name, healthcheck.Retries, ecsParamsHealth.Retries, "healthcheck retries")
+		healthcheck.Timeout = resolveIntPointerResourceOverride(inputCfg.Name, healthcheck.Timeout, ecsParamsHealth.Timeout, "healthcheck timeout")
+		healthcheck.StartPeriod = resolveIntPointerResourceOverride(inputCfg.Name, healthcheck.StartPeriod, ecsParamsHealth.StartPeriod, "healthcheck start_period")
+
+	}
+
+	// validate healthcheck
+	if len(healthcheck.Command) == 0 {
+		return healthcheck, fmt.Errorf("Test/Command is a required field for container healthcheck")
+	}
+
+	return healthcheck, nil
+}
+
+func resolveIntResourceOverride(serviceName string, composeVal, ecsParamsVal int64, option string) int64 {
+	if composeVal > 0 && ecsParamsVal > 0 {
+		showResourceOverrideMsg(serviceName, composeVal, ecsParamsVal, option)
+	}
+	if ecsParamsVal > 0 {
+		return ecsParamsVal
+	}
+	return composeVal
+}
+
+func resolveIntPointerResourceOverride(serviceName string, composeVal, ecsParamsVal *int64, option string) *int64 {
+	if composeVal != nil && ecsParamsVal != nil {
+		showResourceOverrideMsg(serviceName, aws.Int64Value(composeVal), aws.Int64Value(ecsParamsVal), option)
+	}
+	if ecsParamsVal != nil {
+		return ecsParamsVal
+	}
+	return composeVal
+}
+
+func resolveStringSliceResourceOverride(serviceName string, composeVal, ecsParamsVal []*string, option string) []*string {
+	if len(composeVal) > 0 && len(ecsParamsVal) > 0 {
+		log.WithFields(log.Fields{
+			"option name":  option,
+			"service name": serviceName,
+		}).Infof("Using ecs-params value as override")
 	}
-	if inputVal > 0 {
-		showResourceOverrideMsg(serviceName, inputVal, ecsVal, option)
+	if len(ecsParamsVal) > 0 {
+		return ecsParamsVal
 	}
-	return ecsVal
+	return composeVal
 }
```

Comment:
I don't like the repetition of these private methods; feels like we should be able to maybe leverage the empty interface to just have one method for this. Not blocking, but just ear-marking for refactoring down the line.

Reply:
I agree, but I couldn't figure out a nice way in go to make it work using interface{} without lots of ugly type casting in the function to figure out what the real type is

Reply:
Could you do a switch statement on the type? E.g. 
```
 switch v.(type)
case string:
   ....
case int:
  ...
```

Reply:
You have to do something like here: https://stackoverflow.com/questions/6372474/how-to-determine-an-interface-values-real-type

And then the function will return `interface{}` and the code that uses the function will have to cast to the correct type. IMO, this is ugly and harder to read and I prefer having the separate functions that I created for now.